### PR TITLE
Add new commands for DR pre-requisite

### PR DESCRIPTION
**Pull request by umangachapagain**

### CODE REVIEW

**Review by travisn — COMMENTED** [review]
Please also define the command(s) in the main readme.md, and the subtopics such as set.md. This will help understand the commands being reviewed, thanks!

File: docs/set.md
Comment (travisn) [inline docs/set.md:81]: This command simply sets the property on the StorageCluster CR? The same thing could be done with an `oc` command, so I would hesitate to have it in the ODF CLI unless there is a very clear need. It it only to simplify the CLI command compared to an `oc` command to patch the StorageCluster CR?

Comment (umangachapagain) [reply docs/set.md]: Removed this command.

File: cmd/odf/get/dr_prereq.go
Comment (subhamkrai) [inline cmd/odf/get/dr_prereq.go:28]: @umangachapagain can we move implementation details of `Run:` to under `pkg/` folder to make `cmd/` more clean? Thanks

**Review by subhamkrai — CHANGES_REQUESTED** [review]

File: cmd/odf/get/dr_prereq.go
Comment (umangachapagain) [reply cmd/odf/get/dr_prereq.go:14]: Moved.

File: cmd/odf/get/dr_prereq.go
Comment (travisn) [inline cmd/odf/get/dr_prereq.go:13]: In the readme the arg is called a cluster name, rather than an ID. Should one be renamed for consistency?

File: pkg/odf/dr/dr.go
Comment (travisn) [inline pkg/odf/dr/dr.go:19]: The submariner operator is always in this ns?

File: pkg/odf/dr/dr.go
Comment (travisn) [inline pkg/odf/dr/dr.go:86]: Every failure for overlapping subnets just indicates globalnet is required? No other kind of failure?

**Review by travisn — CHANGES_REQUESTED** [review]
Just a few questions or nits

File: pkg/odf/dr/dr.go
Comment (umangachapagain) [reply pkg/odf/dr/dr.go:19]: Yes

File: cmd/odf/get/dr_prereq.go
Comment (umangachapagain) [reply cmd/odf/get/dr_prereq.go:13]: Agreed. Changing this to `<PeerManagedClusterName>`

File: pkg/odf/dr/dr.go
Comment (umangachapagain) [reply pkg/odf/dr/dr.go:86]: Yes. Error is only returned when overlapping subnets are found.

File: cmd/odf/root/root.go
Comment (subhamkrai) [inline cmd/odf/root/root.go:121]:
```suggestion
	client, err := ctrl.New(ClientSets.KubeConfig, ctrl.Options{
		Scheme: scheme,
	})
	if err != nil {
		logging.Fatal(err)
	}
	return client
}
```

@umangachapagain can we move this implementation inside `getClientsets` similar to other clients?

File: pkg/odf/dr/dr.go
Comment (subhamkrai) [reply pkg/odf/dr/dr.go:19]: do we want to cli option for this to overwrite?

**Review by subhamkrai — CHANGES_REQUESTED** [review]
small nits, overall looks good

**Review by travisn — APPROVED** [review]
LGTM after Subham's final questions are answered

**Comment by travisn** [comment]
Ugh, the bot merged it after a single approval even though there were still outstanding requests. I guess we'll need a follow-up PR, thanks.

File: pkg/odf/dr/dr.go
Comment (umangachapagain) [reply pkg/odf/dr/dr.go:19]: I don't see the need as this is hardcoded within submariner project as well.
We can assess it when the need arises.

File: cmd/odf/root/root.go
Comment (umangachapagain) [reply cmd/odf/root/root.go:121]: controller-runtime clients are implemented differently from other ClientSets. We could even replace all the clients in the ClientSets with this single client. That is why I chose to keep it separate (it is not a ClientSet type).

I also looked at replacing other clientsets with this client to keep the dependencies minimal and easy to manage. But, the refactor was needed in the rook-ceph CLI as well (due to hard dependency between the projects). Given controller-runtime client is recommended over ClientSets, should we do a refactor in that direction? @travisn @subhamkrai

Comment (subhamkrai) [reply cmd/odf/root/root.go:121]: > controller-runtime clients are implemented differently from other ClientSets. We could even replace all the clients in the ClientSets with this single client. That is why I chose to keep it separate (it is not a ClientSet type).
> 
> I also looked at replacing other clientsets with this client to keep the dependencies minimal and easy to manage. But, the refactor was needed in the rook-ceph CLI as well (due to hard dependency between the projects). Given controller-runtime client is recommended over ClientSets, should we do a refactor in that direction? @travisn @subhamkrai

@umangachapagain yes, sounds good to do the refactoring.

Comment (subhamkrai) [reply cmd/odf/root/root.go:121]: Will you open a follow-up PR for this?

Comment (umangachapagain) [reply cmd/odf/root/root.go:121]: Sure. Might take a while, but will do.

Comment (subhamkrai) [reply cmd/odf/root/root.go:121]: sure, Thanks

**Pull request merged**
